src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -65,6 +66,21 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.STR_DPKG_WAS_INTERRUPTED = "E: dpkg was interrupted, you must manually run 'sudo dpkg --configure -a' to correct the problem."
         self.ESM_MARKER = "The following packages could receive security updates with UA Infra: ESM service enabled:"
 
+        # pro client pre-requisite checks.
+        try:
+            self.maximum_os_version_supported = 18  # Currently we support only ubuntu 18 or lesser versions.
+            self.__pro_client_prereq_met = False
+            self.ubuntu_pro_client = UbuntuProClient(env_layer, composite_logger)
+
+            # Add condition to check if python3 is also installed.
+            if Constants.PRO_CLIENT_ENABLED and self.__get_os_major_version() < self.maximum_os_version_supported:
+                self.ubuntu_pro_client.install_or_update_pro()
+
+            # This flag will be used to determine if pro client can be used for querying reboot status or packages list.
```

Comment:
Please use the fully qualified name in comments, code and any strings that go up the stack -- so Ubuntu Pro Client vs. just pro client.